src/language-js/print/call-arguments.js

```diff
@@ -314,11 +313,40 @@ function isHopefullyShortCallArgument(node) {
   return isRegExpLiteral(node) || isSimpleCallArgument(node);
 }
 
-function isReactHookCallWithDepsArray(args) {
+/**
+ * Gets `"foo"` from `foo()`.
+ * Gets `"foo"` from `bar.foo()`
+ * Gets `"foo"` from `bar.baz.foo()`
+ * @param {any} callExpression
+ * @returns {(string | null)}
+ */
+function getCallName(callExpression) {
+  const { callee } = callExpression;
+  switch (callee.type) {
+    case "Identifier":
+      return callee.name;
+    case "MemberExpression": {
+      const { property } = callee;
+      if (property.type === "Identifier") {
+        return property.name;
+      }
+    }
+  }
+  return null;
+}
+
+function isReactHookCallWithDepsArray(node) {
+  const callName = node.type === "CallExpression" && getCallName(node);
+  // The name of React Hooks is starts with "use"
+  // https://reactjs.org/docs/hooks-custom.html#extracting-a-custom-hook
+  //   > A custom Hook is a JavaScript function whose name starts with
+  //   > ”use” and that may call other Hooks.
+  const isHookCall = Boolean(callName && callName.startsWith("use"));
```

Comment:
There are also many `use...` functions in this lib too https://vueuse.org/functions.html, I nerver used them and I nerver used React either, so I'm not sure if we want list all those React method names https://beta.reactjs.org/reference/react . If not, we should add somecode from vueuse.

Reply:
Maybe we can simply mke it print like this when the function call has two arguments, and the first is a function expression and the second argument is array literal instead of checking function names?

Reply:
Just some thoughts from a react user here. Adding names would mean painful maintenance later + we can create custom hooks that won't be formatted in this case. The check of a func args, on the other hand, seems reliable enough (unless someone has funcs that are not react hooks with the same arg types, or any other library with the same arg types).